game/world/objects/npc.cpp

```diff
@@ -3234,23 +3280,43 @@ bool Npc::rotateTo(float dx, float dz, float step, bool noAnim, uint64_t dt) {
   float a  = angleDir(dx,dz);
   float da = a-angle;
 
-  if(noAnim || std::cos(double(da)*M_PI/180.0)>0) {
+  if(anim == AnimationSolver::TurnType::None || std::cos(double(da)*M_PI/180.0)>0) {
     if(float(std::abs(int(da)%180))<=(step*2.f)) {
       setAnimRotate(0);
       setDirection(a);
       return false;
       }
     } else {
-    stopWalkAnimation();
+    visual.stopWalkAnim(*this);
     }
 
   const auto sgn = std::sin(double(da)*M_PI/180.0);
   if(sgn<0) {
-    setAnimRotate(noAnim ? 0 : +1);
+    switch(anim) {
+      case AnimationSolver::TurnType::Std:
+        setAnimRotate(+1);
+        break;
+      case AnimationSolver::TurnType::None:
+        setAnimRotate(0);
+        break;
+      case AnimationSolver::TurnType::Whirl:
+        visual.setAnimWhirl(*this,+1);
+        break;
+      }
     setDirection(angle-step);
     } else
   if(sgn>0) {
-    setAnimRotate(noAnim ? 0 : -1);
+    switch(anim) {
```

Comment:
This code might be simplified to cover both positive and negative `sign`, with single switch-case

Reply:
I've adapted the code now to only use one switch